controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -65,47 +97,460 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+		r.cephClientNode.Namespace = r.namespace
+	}
+}
+
 func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
 	log := ctrllog.FromContext(ctx)
 	log.Info("Running StorageClassClaim controller in Consumer Mode")
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
-	log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
+func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Result, error) {
+	r.log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
+
+	r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateCreating)
+
+	for _, cephResourceSpec := range r.storageClassClaim.Status.CephResources {
+		r.cephResourcesByName[cephResourceSpec.Name] = cephResourceSpec
+	}
+
+	if r.storageClassClaim.GetDeletionTimestamp().IsZero() {
+		if !controllers.Contains(r.storageClassClaim.GetFinalizers(), storageClassClaimFinalizer) {
+			r.log.Info("Finalizer not found for StorageClassClaim. Adding finalizer.", "StorageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+			r.storageClassClaim.SetFinalizers(append(r.storageClassClaim.GetFinalizers(), storageClassClaimFinalizer))
+			if err := r.update(r.storageClassClaim); err != nil {
+				r.log.Error(err, "Failed to update StorageClassClaim with finalizer.", "StorageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+				return reconcile.Result{}, err
+			}
+		}
+		if r.storageClassClaim.Spec.Type == "blockpool" {
+
+			if err := r.reconcileCephClientRBDProvisioner(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephClientRBDNode(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephBlockPool(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+		} else if r.storageClassClaim.Spec.Type == "sharedfilesystem" {
+			if err := r.reconcileCephClientCephFSProvisioner(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephClientCephFSNode(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephFilesystemSubVolumeGroup(); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		cephResourcesReady := true
+		for _, cephResource := range r.storageClassClaim.Status.CephResources {
+			if cephResource.Phase != "Ready" {
+				cephResourcesReady = false
+				break
+			}
+		}
+
+		if cephResourcesReady {
+			r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateReady)
+		}
+
+	} else {
+		// todo deletion of the Ceph Resources
+	}
+
+	r.log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
```

Comment:
This is already logged at the start of the function, please remove

```suggestion
```

Reply:
ack